<!--: spam
Content-Type: text/html

<body bgcolor="#f0f0f8"><font color="#f0f0f8" size="-5"> -->
<body bgcolor="#f0f0f8"><font color="#f0f0f8" size="-5"> --> -->
</font> </font> </font> </script> </object> </blockquote> </pre>
</table> </table> </table> </table> </table> </font> </font> </font><body bgcolor="#f0f0f8">
<table width="100%" cellspacing=0 cellpadding=2 border=0 summary="heading">
<tr bgcolor="#6622aa">
<td valign=bottom>&nbsp;<br>
<font color="#ffffff" face="helvetica, arial">&nbsp;<br><big><big><strong>&lt;type 'exceptions.IOError'&gt;</strong></big></big></font></td
><td align=right valign=bottom
><font color="#ffffff" face="helvetica, arial">Python 2.7.10: /usr/bin/python<br>Mon Oct 24 15:07:54 2016</font></td></tr></table>
    
<p>A problem occurred in a Python script.  Here is the sequence of
function calls leading up to the error, in the order they occurred.</p>
<table width="100%" cellspacing=0 cellpadding=0 border=0>
<tr><td bgcolor="#d8bbff"><big>&nbsp;</big><a href="file:///Library/WebServer/Documents/script/heatmap.py">/Library/WebServer/Documents/script/heatmap.py</a> in <strong><module></strong>()</td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;73</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fig.savefig(&nbsp;"imgGen/"+unique_filename+".png",&nbsp;format='png'&nbsp;)<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;74</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#&nbsp;import&nbsp;shutil<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;75</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#&nbsp;shutil.copyfileobj(open("tempfile.png",'rb'),&nbsp;sys.stdout)<br>
</tt></font></td></tr>
<tr><td bgcolor="#ffccee"><tt>=&gt;<small>&nbsp;&nbsp;&nbsp;76</small>&nbsp;make_fig()<br>
</tt></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;77</small>&nbsp;crop("imgGen/"+unique_filename+".png")<br>
</tt></font></td></tr>
<tr><td><small><font color="#909090"><strong>make_fig</strong>&nbsp;= &lt;function make_fig&gt;</font></small></td></tr></table>
<table width="100%" cellspacing=0 cellpadding=0 border=0>
<tr><td bgcolor="#d8bbff"><big>&nbsp;</big><a href="file:///Library/WebServer/Documents/script/heatmap.py">/Library/WebServer/Documents/script/heatmap.py</a> in <strong>make_fig</strong>()</td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;33</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;filename,&nbsp;file_extension&nbsp;=&nbsp;os.path.splitext(filecsv)<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;34</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;#&nbsp;print&nbsp;sys.argv[1]<br>
</tt></font></td></tr>
<tr><td bgcolor="#ffccee"><tt>=&gt;<small>&nbsp;&nbsp;&nbsp;35</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;x,&nbsp;y&nbsp;=&nbsp;np.genfromtxt(filecsv,&nbsp;delimiter=',',&nbsp;unpack=True)<br>
</tt></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;36</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;form&nbsp;=&nbsp;cgi.FieldStorage()<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;&nbsp;37</small>&nbsp;<br>
</tt></font></td></tr>
<tr><td><small><font color="#909090">x <em>undefined</em>, y <em>undefined</em>, <em>global</em> <strong>np</strong>&nbsp;= &lt;module 'numpy' from '/System/Library/Frameworks...rsions/2.7/Extras/lib/python/numpy/__init__.pyc'&gt;, np.<strong>genfromtxt</strong>&nbsp;= &lt;function genfromtxt&gt;, <strong>filecsv</strong>&nbsp;= '4.csv', delimiter <em>undefined</em>, unpack <em>undefined</em>, <em>builtin</em> <strong>True</strong>&nbsp;= True</font></small></td></tr></table>
<table width="100%" cellspacing=0 cellpadding=0 border=0>
<tr><td bgcolor="#d8bbff"><big>&nbsp;</big><a href="file:///System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/lib/npyio.py">/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/lib/npyio.py</a> in <strong>genfromtxt</strong>(fname='4.csv', dtype=&lt;type 'float'&gt;, comments='#', delimiter=',', skiprows=0, skip_header=0, skip_footer=0, converters=None, missing='', missing_values=None, filling_values=None, usecols=None, names=None, excludelist=None, deletechars=None, replace_space='_', autostrip=False, case_sensitive=True, defaultfmt='f%i', unpack=True, usemask=False, loose=True, invalid_raise=True)</td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;1329</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;try:<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;1330</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if&nbsp;isinstance(fname,&nbsp;basestring):<br>
</tt></font></td></tr>
<tr><td bgcolor="#ffccee"><tt>=&gt;<small>&nbsp;1331</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fhd&nbsp;=&nbsp;iter(np.lib._datasource.open(fname,&nbsp;'rbU'))<br>
</tt></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;1332</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;own_fhd&nbsp;=&nbsp;True<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;1333</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;else:<br>
</tt></font></td></tr>
<tr><td><small><font color="#909090">fhd <em>undefined</em>, <em>builtin</em> <strong>iter</strong>&nbsp;= &lt;built-in function iter&gt;, <em>global</em> <strong>np</strong>&nbsp;= &lt;module 'numpy' from '/System/Library/Frameworks...rsions/2.7/Extras/lib/python/numpy/__init__.pyc'&gt;, np.<strong>lib</strong>&nbsp;= &lt;module 'numpy.lib' from '/System/Library/Framew...ns/2.7/Extras/lib/python/numpy/lib/__init__.pyc'&gt;, np.lib.<strong>_datasource</strong>&nbsp;= &lt;module 'numpy.lib._datasource' from '/System/Li...2.7/Extras/lib/python/numpy/lib/_datasource.pyc'&gt;, np.lib._datasource.<strong>open</strong>&nbsp;= &lt;function open&gt;, <strong>fname</strong>&nbsp;= '4.csv'</font></small></td></tr></table>
<table width="100%" cellspacing=0 cellpadding=0 border=0>
<tr><td bgcolor="#d8bbff"><big>&nbsp;</big><a href="file:///System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/lib/_datasource.py">/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/lib/_datasource.py</a> in <strong>open</strong>(path='4.csv', mode='rbU', destpath='.')</td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;145</small>&nbsp;<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;146</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ds&nbsp;=&nbsp;DataSource(destpath)<br>
</tt></font></td></tr>
<tr><td bgcolor="#ffccee"><tt>=&gt;<small>&nbsp;&nbsp;147</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;ds.open(path,&nbsp;mode)<br>
</tt></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;148</small>&nbsp;<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;149</small>&nbsp;<br>
</tt></font></td></tr>
<tr><td><small><font color="#909090"><strong>ds</strong>&nbsp;= &lt;numpy.lib._datasource.DataSource object&gt;, ds.<strong>open</strong>&nbsp;= &lt;bound method DataSource.open of &lt;numpy.lib._datasource.DataSource object&gt;&gt;, <strong>path</strong>&nbsp;= '4.csv', <strong>mode</strong>&nbsp;= 'rbU'</font></small></td></tr></table>
<table width="100%" cellspacing=0 cellpadding=0 border=0>
<tr><td bgcolor="#d8bbff"><big>&nbsp;</big><a href="file:///System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/lib/_datasource.py">/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/lib/_datasource.py</a> in <strong>open</strong>(self=&lt;numpy.lib._datasource.DataSource object&gt;, path='4.csv', mode='rbU')</td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;492</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;_file_openers[ext](found,&nbsp;mode=mode)<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;493</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;else:<br>
</tt></font></td></tr>
<tr><td bgcolor="#ffccee"><tt>=&gt;<small>&nbsp;&nbsp;494</small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raise&nbsp;IOError("%s&nbsp;not&nbsp;found."&nbsp;%&nbsp;path)<br>
</tt></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;495</small>&nbsp;<br>
</tt></font></td></tr>
<tr><td><font color="#909090"><tt>&nbsp;&nbsp;<small>&nbsp;&nbsp;496</small>&nbsp;<br>
</tt></font></td></tr>
<tr><td><small><font color="#909090"><em>builtin</em> <strong>IOError</strong>&nbsp;= &lt;type 'exceptions.IOError'&gt;, <strong>path</strong>&nbsp;= '4.csv'</font></small></td></tr></table><p><strong>&lt;type 'exceptions.IOError'&gt;</strong>: 4.csv not found.
<br><tt><small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</small>&nbsp;</tt>args&nbsp;=
('4.csv not found.',)
<br><tt><small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</small>&nbsp;</tt>errno&nbsp;=
None
<br><tt><small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</small>&nbsp;</tt>filename&nbsp;=
None
<br><tt><small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</small>&nbsp;</tt>message&nbsp;=
'4.csv not found.'
<br><tt><small>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</small>&nbsp;</tt>strerror&nbsp;=
None


<!-- The above is a description of an error in a Python program, formatted
     for a Web browser because the 'cgitb' module was enabled.  In case you
     are not reading this in a Web browser, here is the original traceback:

Traceback (most recent call last):
  File "heatmap.py", line 76, in &lt;module&gt;
    make_fig()
  File "heatmap.py", line 35, in make_fig
    x, y = np.genfromtxt(filecsv, delimiter=',', unpack=True)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/lib/npyio.py", line 1331, in genfromtxt
    fhd = iter(np.lib._datasource.open(fname, 'rbU'))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/lib/_datasource.py", line 147, in open
    return ds.open(path, mode)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/lib/_datasource.py", line 494, in open
    raise IOError("%s not found." % path)
IOError: 4.csv not found.

-->

